e2e/app/create-account-page.ts

```diff
@@ -0,0 +1,226 @@
+    for (const element of elements) {
+      const innerTxt  = await (await element.getProperty('innerText')).jsonValue();
+      if (innerTxt === checkboxLabel) {
+        return element;
+      }
+    }
+
+  }
+
+  public async getInstitutionNameInput(): Promise<ElementHandle> {
+    return await this.puppeteerPage.waitForXPath('//input[@placeholder=\'Institution Name\']');
+  }
+
+  public async getResearchBackgroundTextarea(): Promise<ElementHandle> {
+    const label = 'Please describe your research background, experience and research interests';
+    return await this.puppeteerPage.waitForXPath(`//label[contains(normalize-space(.),'${label}')]/parent::*//textarea`)
+  }
+
+  public async fillInFormFields(fields: Array<{ label: string; value: string; }>): Promise<string> {
+    let newUserName;
+
+    function formFieldXpathHelper(aLabel: string): string {
+      return `//label[contains(normalize-space(.),'${aLabel}')]/parent::*//input[@type='text']`;
+    }
+
+    for (const field of fields) {
+      const selector = formFieldXpathHelper(field.label);
+      const e = await this.puppeteerPage.waitForXPath(selector, {visible: true});
+      await e.focus();
+      await e.type(field.value);
+      await e.press('Tab', { delay: 100 }); // tab out
+      if (field.label === 'New Username') {
+        await this.puppeteerPage.waitForSelector('clr-icon.is-solid[shape="success-standard"]', {visible: true});
+        newUserName = field.value;
+      }
+    }
+    return newUserName;
+  }
+
+  // select Institution Affiliation from a dropdown
+  public async selectInstitution(selectTextValue: string) {
+    const dropdown = new DropdownSelect(this.puppeteerPage);
+    await dropdown.select(selectTextValue);
+  }
+
+  public async getInstitutionValue() {
+    const dropdown = new DropdownSelect(this.puppeteerPage);
+    return await dropdown.displayedValue();
+  }
+
+  // select Institution Affiliation from a dropdown
+  public async selectEducationLevel(selectTextValue: string) {
+    const dropdown = new DropdownSelect(this.puppeteerPage, 'Highest Level of Education Completed');
+    await dropdown.select(selectTextValue);
+  }
+
+  // select Year of Birth from a dropdown
+  public async selectYearOfBirth(year: string) {
+    const dropdown = new DropdownSelect(this.puppeteerPage, 'Year of Birth');
+    await dropdown.select(year);
+  }
+
+  // Combined steps to make test code cleaner and shorter
+
+  // Step 1: Enter Invitation key
+  public async fillOutInvitationKey() {
+    await this.getInvitationKeyInput()
+    .then(invitationKeyInput => invitationKeyInput.type(process.env.INVITATION_KEY))
+    .then(() => this.getNextButton())
+    .then(submitButton => submitButton.click());
+  }
+
+  // Step 2: Accepting Terms of Use and Privacy statement.
+  public async acceptTermsOfUseAgreement() {
+    const privacyStatementCheckbox = await this.getPrivacyStatementCheckbox();
+    expect(privacyStatementCheckbox.asElement()).toBeTruthy();
```

Comment:
[non-opt / requires response] Personally I think all 'expect' calls should live in test classes, not in the page object. Was there a particular reason you opted for this design?

If you wanted to break this out into a method, I would make it a private method on the test class, rather than part of the page component itself. It just feels too surprising to have test expectations living in a page object like this, at least on first look.

Reply:
agree to remove `expect` in page file. I didn't like it myself.